src/main/java/org/jenkinsci/plugins/github_branch_source/RepositoryUriResolver.java

```diff
@@ -38,17 +40,14 @@
 
     public static String hostnameFromApiUri(String apiUri) {
         apiUri = Util.fixEmptyAndTrim(apiUri);
-        if (apiUri == null) {
-            return "github.com";
+        try {
+            URL endpoint = new URL(apiUri);
```

Comment:
:ant: it is not nice to pass a known null value to the `URL` constructor and rely on that throwing `MalformedURLException`. Better to be clear and skip the host check in this case.